Cannot upgrade from 8.0.4 or 9.3.2 to 9.4.0 alpha

## Description of bug
Cannot login into the upgrade wizard when trying to upgrade from Dnn 8.0.4 to upgrade to 9.4.0 alpha

## Steps to reproduce
List the steps to reproduce the behavior:
1. Install a clean Dnn 8.0.4 instance
2. Overwrite the files with Dnn 9.4.0 upgrade zip file
3. Fire up the site
4. When the upgrade wizard comes up, try to login as host

## Current result
"Username or password not recognized" error shows up

## Expected result
Upgrade wizard should start

## Error log
In the log4net logs, I have this error:
```
2019-05-28 16:42:32,153 [DanielValadasPC][Thread:20][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: TermsTabId
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Entities.Portals.PortalInfo.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)
2019-05-28 16:42:44,580 [DanielValadasPC][Thread:6][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: HasAgreedToTerms
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Security.Membership.AspNetMembershipProvider.FillUserInfo(Int32 portalId, IDataReader dr, Boolean closeDataReader)
   at DotNetNuke.Security.Membership.AspNetMembershipProvider.GetUserByUserNameFromDataStore(Int32 portalId, String username)
   at DotNetNuke.Security.Membership.AspNetMembershipProvider.<>c__DisplayClass73_0.<GetUserByUserName>b__0(CacheItemArgs _)
   at DotNetNuke.Common.Utilities.DataCache.GetCachedDataFromRuntimeCache(CacheItemArgs cacheItemArgs, CacheItemExpiredCallback cacheItemExpired)
```

## Affected version

* [x] 9.4.0 alpha
* [ ] 9.3.2
* [ ] 9.3.1
* [ ] 9.2.2
* [ ] 9.2.1
* [ ] 9.2
* [ ] 9.1.1
* [ ] 9.1
* [ ] 9.0

--- a/DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs
+++ b/DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs
@@ -512,10 +512,7 @@ namespace DotNetNuke.Security.Membership
                         IsSuperUser = Null.SetNullBoolean(dr["IsSuperUser"]),
                         UserID = Null.SetNullInteger(dr["UserID"]),
                         DisplayName = Null.SetNullString(dr["DisplayName"]),
-                        LastIPAddress = Null.SetNullString(dr["LastIPAddress"]),
-                        HasAgreedToTerms = Null.SetNullBoolean(dr["HasAgreedToTerms"]),
-                        HasAgreedToTermsOn = Null.SetNullDateTime(dr["HasAgreedToTermsOn"]),
-                        RequestsRemoval = Null.SetNullBoolean(dr["RequestsRemoval"])
+                        LastIPAddress = Null.SetNullString(dr["LastIPAddress"])
                     };
 
                     var schema = dr.GetSchemaTable();
@@ -529,16 +526,22 @@ namespace DotNetNuke.Security.Membership
                         {
                             user.VanityUrl = Null.SetNullString(dr["VanityUrl"]);
                         }
-                    }
-
-                    user.AffiliateID = Null.SetNullInteger(Null.SetNull(dr["AffiliateID"], user.AffiliateID));
-                    user.Username = Null.SetNullString(dr["Username"]);
-                    UserController.GetUserMembership(user);
-                    user.Email = Null.SetNullString(dr["Email"]);
-                    user.Membership.UpdatePassword = Null.SetNullBoolean(dr["UpdatePassword"]);
-
-                    if (schema != null)
-                    {
+                        if (schema.Select("ColumnName = 'HasAgreedToTerms'").Length > 0)
+                        {
+                            user.HasAgreedToTerms = Null.SetNullBoolean(dr["HasAgreedToTerms"]);
+                        }
+                        if (schema.Select("ColumnName = 'HasAgreedToTermsOn'").Length > 0)
+                        {
+                            user.HasAgreedToTermsOn = Null.SetNullDateTime(dr["HasAgreedToTermsOn"]);
+                        }
+                        else
+                        {
+                            user.HasAgreedToTermsOn = Null.NullDate;
+                        }
+                        if (schema.Select("ColumnName = 'RequestsRemoval'").Length > 0)
+                        {
+                            user.RequestsRemoval = Null.SetNullBoolean(dr["RequestsRemoval"]);
+                        }
                         if (schema.Select("ColumnName = 'PasswordResetExpiration'").Length > 0)
                         {
                             user.PasswordResetExpiration = Null.SetNullDateTime(dr["PasswordResetExpiration"]);
@@ -548,6 +551,13 @@ namespace DotNetNuke.Security.Membership
                             user.PasswordResetToken = Null.SetNullGuid(dr["PasswordResetToken"]);
                         }
                     }
+
+                    user.AffiliateID = Null.SetNullInteger(Null.SetNull(dr["AffiliateID"], user.AffiliateID));
+                    user.Username = Null.SetNullString(dr["Username"]);
+                    UserController.GetUserMembership(user);
+                    user.Email = Null.SetNullString(dr["Email"]);
+                    user.Membership.UpdatePassword = Null.SetNullBoolean(dr["UpdatePassword"]);
+
                     if (!user.IsSuperUser)
                     {
                         user.Membership.Approved = Null.SetNullBoolean(dr["Authorised"]);

